core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -395,7 +397,9 @@ public static String stripInstanceAndHost(String metricsName) {
    * @return the string representing the escaped URI
    */
   public static String escape(AlluxioURI uri) {
-    return uri.toString().replace("%", "%25").replace("/", "%2F").replace(".", "%2E");
+    return CACHED_ESCAPED_PATH.computeIfAbsent(uri,
+        u -> u.toString().replace("%", "%25")
+            .replace("/", "%2F").replace(".", "%2E"));
```

Comment:
If you're looking to escape into URI-compatible paths I would look somewhere in apache commons utils

Reply:
I haven't found existing utils doing exactly the same thing. If there is a util method, that would be really convenient.

Reply:
I don't know what the original intention of this code was (if it is doing a URL encoding on the metric name or not, but it looks like it to me.

We should be using this: https://docs.oracle.com/javase/8/docs/api/java/net/URLEncoder.html

Reply:
I think for the first time `MetricsSystem.escape` created for Metric name processing. Like we use `.` for distinguish metric name components like (`InstanceType.MetricName.Tag:TagValue`), so the given url should not have `.`.

This use case does not fit in `UTLEncoder` because it will `The special characters ".", "-", "*", and "_" remain the same.`

Reply:
Since the code was already here I guess it's ok - but this seems very sketchy to me...we should improve it in the future.

Reply:
Yeah, it's sketchy, will reconsider how to improve it in the future